x/rewards/client/cli/query_generate_ibc_iprpc_tx.go

```diff
@@ -0,0 +1,213 @@
+		 - "--node" flag which should be the node from which the ibc-transfer TX originates (probably not a Lava node).
+		
+		To generate only the memo, so it can be embedded manually when sending an ibc-transfer, use the optional 
+		--memo-only flag (in this case, the amount, src-port, and src-channel arguments will be ignored).`,
+		Args: cobra.RangeArgs(2, 5),
+		Example: `
+		lavad q rewards generate-ibc-iprpc-tx ETH1 3 100ulava transfer channel-0 --from bob --node tcp://localhost:36657 > tx.json
+		lavad tx sign tx.json --from bob > signedTx.json
+		lavad tx broadcast signedTx.json --from bob --node tcp://localhost:36657
+		`,
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			spec := args[0]
+			duration := args[1]
+
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			clientCtx.GenerateOnly = true
+			sender := clientCtx.FromAddress.String()
+
+			// create memo
+			senderAddr, err := utils.ParseCLIAddress(clientCtx, sender)
+			if err != nil {
+				return err
+			}
+			memo, err := createIbcIprpcMemo(senderAddr, spec, duration)
+			if err != nil {
+				return err
+			}
+
+			// handle --memo-only
+			memoOnly, err := cmd.Flags().GetBool(FlagMemoOnly)
+			if err != nil {
+				return err
+			}
+			if memoOnly {
+				cmd.Println(escapeMemo(memo))
+				return nil
+			}
+
+			// create ibc-transfer msg
+			if len(args) != 5 {
+				return fmt.Errorf("not enough arguments for generating TX")
+			}
+			node, err := cmd.Flags().GetString(flags.FlagNode)
+			if err != nil {
+				return err
+			}
+			clientQueryCtx, err := client.GetClientQueryContext(cmd)
+			if err != nil {
+				return err
+			}
+			clientQueryCtx = clientQueryCtx.WithNodeURI(node)
+			clientCtx = clientCtx.WithNodeURI(node)
+
+			amount, err := sdk.ParseCoinNormalized(args[2])
+			if err != nil {
+				return err
+			}
+			srcPort := args[3]
+			srcChannel := args[4]
+			height, err := getTimeoutHeight(clientQueryCtx, srcPort, srcChannel)
+			if err != nil {
+				return err
+			}
+			msg := ibctransfertypes.NewMsgTransfer(srcPort, srcChannel, amount, sender, types.IbcIprpcReceiverAddress(), height, 0, memo)
+
+			// print constructed transfer message with custom memo
+			txf, err := tx.NewFactoryCLI(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+			fees, err := getFees(clientQueryCtx)
+			if err != nil {
+				return err
+			}
+			txf = txf.WithFees(fees)
+
+			return txf.PrintUnsignedTx(clientCtx, msg)
+		},
+	}
+
+	cmd.Flags().Bool(FlagMemoOnly, false, "Generate only an ibc-transfer packet's memo for IPRPC over IBC")
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.MarkFlagRequired(flags.FlagFrom)
+	cmd.MarkFlagRequired(flags.FlagNode)
+	return cmd
+}
+
+func createIbcIprpcMemo(creator string, spec string, duration string) (string, error) {
+	// sanity check
+	_, err := strconv.ParseUint(duration, 10, 64)
+	if err != nil {
+		return "", err
+	}
+
+	data := map[string]interface{}{
```

Comment:
use the original struct and than marshal it

Reply:
done. see https://github.com/lavanet/lava/commit/e29f0e73644f40f55eb5f8621c5a58df6cef7898